src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -50,6 +49,10 @@ internal sealed class AspNetCoreDiagnosticObserver : DiagnosticObserver
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<AspNetCoreDiagnosticObserver>();
         private readonly Tracer _tracer;
 
+        private readonly IntegrationInfo _integrationId = IntegrationId;
+        private readonly string _requestInOperationName = HttpRequestInOperationName;
+        private readonly Action<Span> _spanTransform = s => { };
```

Comment:
Given we call `_spanTransform?.Invoke(scope.Span)` later, is it worth leaving this null? Or don't, and remove the null check?